Makefile

```diff
@@ -0,0 +1,62 @@
+# if provided, FILTER is used as the --filter argument to `swift test`.
+ifdef FILTER
+	FILTERARG = --filter $(FILTER)
+else
+	FILTERARG =
+endif
+
+ifdef DOCSVERSION
+	DOCSARG = --module-version $(DOCSVERSION)
+else
+	DOCSARG =
+endif
+
+# if no value provided assume sourcery is in the user's PATH
+SOURCERY ?= sourcery
+
+define check_for_gem
+	gem list $(1) -i > /dev/null || gem install $(1) || { echo "ERROR: Failed to locate or install the ruby gem $(1); please install yourself with 'gem install $(1)' (you may need to use sudo)"; exit 1; }
+endef
+
+all:
+	swift build
+
+# project generates the .xcodeproj, and then modifies it to add
+# spec .json files to the project
+project:
+	swift package generate-xcodeproj
+	@$(call check_for_gem,xcodeproj)
+	ruby etc/add_json_files.rb
+
+linuxmain:
+	$(SOURCERY) --sources Tests/ --exclude-sources Tests/DisabledTests/ --templates Tests/LinuxMain.stencil --output Tests/LinuxMain.swift
+	@tail -n +5 Tests/LinuxMain.swift > Tests/LinuxMain.swift.tmp
```

Comment:
what are we doing with tail and the temp file here? think a comment would be helpful for future readers

Reply:
Silly trick here that I think we just want better CI to do, but it removes the Sourcery version text so you don't have to manually change it 😅

Reply:
sgtm, can we add a comment so we remember that in the future?